File: lib/domain/entities/project.dart
class Project {
  final String id;
  final String? parentId;
  final String name;
  final String? description;
  final String color;
  final String icon;
  final DateTime? startDate;
  final DateTime? endDate;
  final DateTime createdAt;
  final bool isDefault;

  const Project({
    required this.id,
    this.parentId,
    required this.name,
    this.description,
    required this.color,
    required this.icon,
    this.startDate,
    this.endDate,
    required this.createdAt,
    this.isDefault = false,
  });

  Project copyWith({
    String? id,
    String? parentId,
    String? name,
    String? description,
    String? color,
    String? icon,
    DateTime? startDate,
    DateTime? endDate,
    DateTime? createdAt,
    bool? isDefault,
  }) {
    return Project(
      id: id ?? this.id,
      parentId: parentId ?? this.parentId,
      name: name ?? this.name,
      description: description ?? this.description,
      color: color ?? this.color,
      icon: icon ?? this.icon,
      startDate: startDate ?? this.startDate,
      endDate: endDate ?? this.endDate,
      createdAt: createdAt ?? this.createdAt,
      isDefault: isDefault ?? this.isDefault,
    );
  }
}

File: backend/src/models/project.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use chrono::{DateTime, Utc, FixedOffset};

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Project {
    pub id: String,
    pub user_id: String,
    pub parent_id: Option<String>,
    pub name: String,
    pub description: Option<String>,
    pub color: String,
    pub icon: String,
    pub start_date: Option<DateTime<FixedOffset>>,
    pub end_date: Option<DateTime<FixedOffset>>,
    pub created_at: DateTime<FixedOffset>,
    pub updated_at: DateTime<FixedOffset>,
    pub sort_order: i32,
    pub is_default: bool,
    pub deleted_at: Option<DateTime<FixedOffset>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateProject {
    pub parent_id: Option<String>,
    pub name: String,
    pub description: Option<String>,
    #[serde(default = "default_color")]
    pub color: String,
    #[serde(default = "default_icon")]
    pub icon: String,
    pub start_date: Option<DateTime<FixedOffset>>,
    pub end_date: Option<DateTime<FixedOffset>>,
    #[serde(default)]
    pub sort_order: i32,
    #[serde(default)]
    pub is_default: bool,
}

fn default_color() -> String {
    "#6366f1".to_string()
}

fn default_icon() -> String {
    "folder".to_string()
}